std/parallelism.d

```diff
@@ -2864,10 +2864,10 @@ public:
             }
         }
 
-        ref T opIndex(size_t index)
+        ref inout(T) opIndex(size_t index) inout
```

Comment:
Andrei clearly doesn't want `inout` at all costs, even though in this case, it's vastly superior and easier to understand. Not sure what else to say aside from what you've already said.